api/src/test/java/com/epam/pipeline/controller/pipeline/PipelineControllerTest.java

```diff
@@ -0,0 +1,827 @@
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldCreateOrRenamePipelineFolder() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(sourceItemVO);
+        doReturn(gitCommitEntry).when(mockPipelineApiService).createOrRenameFolder(ID, sourceItemVO);
+
+        final MvcResult mvcResult = performRequest(post(String.format(PIPELINE_ID_FOLDER_URL, ID)).content(content));
+
+        verify(mockPipelineApiService).createOrRenameFolder(ID, sourceItemVO);
+        assertResponse(mvcResult, gitCommitEntry, GitCreatorUtils.GIT_COMMIT_ENTRY_TYPE);
+    }
+
+    @Test
+    public void shouldFailCreateOrRenamePipelineFolderForUnauthorizedUser() {
+        performUnauthorizedRequest(post(String.format(PIPELINE_ID_FOLDER_URL, ID)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldRemovePipelineFolder() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(sourceItemVO);
+        doReturn(gitCommitEntry).when(mockPipelineApiService).removeFolder(ID, TEST_STRING, TEST_STRING, TEST_STRING);
+
+        final MvcResult mvcResult = performRequest(delete(String.format(PIPELINE_ID_FOLDER_URL, ID)).content(content));
+
+        verify(mockPipelineApiService).removeFolder(ID, TEST_STRING, TEST_STRING, TEST_STRING);
+        assertResponse(mvcResult, gitCommitEntry, GitCreatorUtils.GIT_COMMIT_ENTRY_TYPE);
+    }
+
+    @Test
+    public void shouldFailRemovePipelineFolderForUnauthorizedUser() {
+        performUnauthorizedRequest(delete(String.format(PIPELINE_ID_FOLDER_URL, ID)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldGetPipelineDocs() throws Exception {
+        doReturn(gitRepositoryEntries).when(mockPipelineApiService).getPipelineDocs(ID, TEST_STRING);
+
+        final MvcResult mvcResult = performRequest(get(String.format(PIPELINE_ID_DOCS_URL, ID))
+                .params(multiValueMapOf(VERSION, TEST_STRING)));
+
+        verify(mockPipelineApiService).getPipelineDocs(ID, TEST_STRING);
+        assertResponse(mvcResult, gitRepositoryEntries, GitCreatorUtils.GIT_REPOSITORY_ENTRY_LIST_TYPE);
+    }
+
+    @Test
+    public void shouldFailGetPipelineDocsForUnauthorizedUser() {
+        performUnauthorizedRequest(get(String.format(PIPELINE_ID_DOCS_URL, ID)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldGetPipelineFile() throws Exception {
+        doReturn(TEST_ARRAY).when(mockPipelineApiService).getPipelineFileContents(ID, TEST_STRING, TEST_STRING);
+
+        final MvcResult mvcResult = performRequest(get(String.format(PIPELINE_ID_FILE_URL, ID))
+                .params(multiValueMapOf(VERSION, TEST_STRING,
+                                        PATH, TEST_STRING)));
+        final String contentAsString = mvcResult.getResponse().getContentAsString();
+        final byte[] returnedEntityBody = objectMapper.readValue(contentAsString, byte[].class);
```

Comment:
Could you please extract these lines along with the corresponding assert to a private method in `PipelineControllerTest`.